### PR TITLE
os/mac/version: update method signatures

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?
- [ ] Have you successfully run `brew man` locally and committed any changes?

-----

Follow-up to #10331

### CODE REVIEW

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]
I'm not convinced by the benefits of `typed: strict` over `typed: true`. I really don't see this catching additional bugs and the code becomes pretty much impossible to follow without understanding of all the Sorbet methods. Note that Shopify, the creators of Sorbet, do not seem to be aiming for `typed: strict` (https://shopify.engineering/adopting-sorbet). I'm sure I read that they literally advise against it (but I cannot find the source).

I think it would be preferable to increase the number of `typed: true` files we have and remove `typed: strict` in all cases except those where it requires zero additional changes above `typed: true`.

**Comment by SeekingMeaning** [comment]
> Note that Shopify, the creators of Sorbet, do not seem to be aiming for `typed: strict` (https://shopify.engineering/adopting-sorbet). I'm sure I read that they literally advise against it (but I cannot find the source).

Perhaps `typed: strong` is what you're referring to? https://sorbet.org/docs/static#file-level-granularity-strictness-levels

Regardless, I agree that adding `T.let` everywhere does make the code a bit unreadable and have updated the PR

File: Library/Homebrew/os/mac/version.rb [hunk 1]
```diff
@@ -33,7 +34,7 @@ def self.from_symbol(sym)
         new(str, arch: arch)
       end
 
-      sig { params(value: T.any(String, Symbol)).returns(T::Array[String]) }
+      sig { params(value: T.any(String, Symbol)).returns(T::Array[T.nilable(String)]) }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/os/mac/version.rb:37]: Is this nilable? Could it not be?

**Review by MikeMcQuaid — APPROVED** [review]
Thanks, looks much nicer. Yeh, you're right that I was probably thinking of `strong` here.

File: Library/Homebrew/os/mac/version.rb (hunk 1 above)
Comment (SeekingMeaning) [reply Library/Homebrew/os/mac/version.rb:37]: Yes, the second element, `arch`, can be nil

We could also do:

```rb
returns(T.any([], [String, T.nilable(String)]))
```

but I'm not sure if that's an improvement

Comment (MikeMcQuaid) [reply Library/Homebrew/os/mac/version.rb:37]: Yeh, that feels like a mild improvement 👍🏻

**Pull request merged**